src/test/scala/com/codacy/parsers/CloverParserTest.scala

```diff
@@ -3,124 +3,189 @@ package com.codacy.parsers
-  private val coberturaReportPath = "src/test/resources/test_cobertura.xml"
-  private val cloverReportPath = "src/test/resources/test_clover.xml"
-  private val cloverWithoutPackagesFilePath = "src/test/resources/test_clover_without_packages.xml"
-  private val nonExistentReportPath = "src/test/resources/non-existent.xml"
-  private val invalidCloverReportPath = "src/test/resources/test_invalid_clover.xml"
+class CloverParserTest extends WordSpec with Matchers with EitherValues {
 
   "parse" should {
-    "identify report as invalid" when {
-      "file does not follow the Cobertura format" in {
-        // use some coverage file that does not follow the Clover format
-        val reader = CloverParser.parse(new File("."), new File(coberturaReportPath))
 
-        reader shouldBe 'left
+    "fail to parse an invalid report" when {
+
+      "the report file does not exist" in {
+        // Arrange
+        val nonExistentReportPath = "src/test/resources/non-existent.xml"
+
+        // Act
+        val parseResult = CloverParser.parse(new File("."), new File(nonExistentReportPath))
+
+        // Assert
+        parseResult shouldBe 'left
       }
 
-      "file does not exist" in {
-        val reader = CloverParser.parse(new File("."), new File(nonExistentReportPath))
+      "the report is not in the Clover format" in {
+        // Arrange
+        val reportNotInCloverFormat = "src/test/resources/test_cobertura.xml"
+
+        // Act
+        val parseResult = CloverParser.parse(new File("."), new File(reportNotInCloverFormat))
 
-        reader shouldBe 'left
+        // Assert
+        parseResult shouldBe 'left
       }
 
-      "file does not follow the Clover schema" in {
-        val reader = CloverParser.parse(new File("."), new File(invalidCloverReportPath))
-        reader shouldBe 'left
+      "the report is missing the statements attribute in the file metrics element" in {
+        // Arrange
+        val invalidCloverReportPath = "src/test/resources/test_invalid_clover.xml"
+
+        // Act
+        val parseResult = CloverParser.parse(new File("."), new File(invalidCloverReportPath))
+
+        // Assert
+        parseResult shouldBe 'left
       }
+
     }
 
-    "identify the report as valid" when {
-      "report has packages" in {
-        val reader = CloverParser.parse(new File("."), new File(cloverReportPath))
-        reader shouldBe 'right
+    val cloverReportPath = "src/test/resources/test_clover.xml"
+    val cloverWithoutPackagesFilePath = "src/test/resources/test_clover_without_packages.xml"
+
+    "succeed to parse a valid report" when {
+
+      "the report has packages" in {
+        // Act
+        val parseResult = CloverParser.parse(new File("."), new File(cloverReportPath))
+
+        // Assert
+        parseResult shouldBe 'right
       }
 
-      "report does not have packages" in {
-        val readerWithoutPackages = CloverParser
+      "the report does not have packages" in {
```

Comment:
The test naming seems a bit strange, since it seems it should always succeed, whether it has packages or not. Maybe this name could be improved to something like: "the report does not have packages but has X"

Reply:
Not sure if that makes sense. In this case it seems like the element/tag `<package>` is optional so in this scenario it just doesn't exist.